File: new blog/Blog.Data/Entities/Post.cs
using Microsoft.AspNetCore.Identity;
using System;
using System.Collections;
using System.Collections.Generic;

namespace Blog.Data.Entities
{
    public class Post
    {
        public int PostId { get; set; }

        public string Title { get; set; }

        public string Content { get; set; }

        public DateTime DateCreated { get; set; }

        public int CategoryId { get; set; }
        public Category Category { get; set; }

        public string AuthorId { get; set; }
        public ApplicationUser Author { get; set; }

        public ICollection<Comment> Comments { get; set; }
    }
}

File: FrontEnd/src/app/models/post.ts
import { IComment } from "./comment";
import { ICategory } from "./category";
import { IUser } from "./user";

export interface IPost {
    postId: number;
    title: string;
    content: string;
    dateCreated: Date;
    comments: IComment[];
    category: ICategory;
    author: IUser;
}